File: dotnet/cafdemalihapa/Directories/Hosting/Feature/Automations/EnvironmentVariablesTemplateFiles/DirectoriesMultitude/DirectoriesMultitudeCommandingOrderRectoActionShut.cs
using System.Text;
using cafdemalihapa.Directories.Hosting.Feature.BackEnd;
using cafdemalihapa.Directories.Hosting.Feature.Calls;
using cafdemalihapa.Directories.Hosting.Feature.FrontEnd.FrontEndGuest;
using cafdemalihapa.Directories.Hosting.Feature.FrontEnd.FrontEndPrimary;
using cafdemalihapa.Directories.Hosting.Feature.NotesAndMessages;
using cafdemalihapa.Directories.Hosting.Feature.Tools;
using cafdemalihapa.Directories.Hosting.Feature.WebLinks;
using cafdemalihapa.Helpers;
using Microsoft.Extensions.Logging;

namespace cafdemalihapa.Directories.Hosting.Feature.Automations.EnvironmentVariablesTemplateFiles.DirectoriesMultitude
{
    public class DirectoriesMultitudeCommandingOrderRectoActionShut
    (
        StringHelpers stringHelpers,
        Operations.OperationsDirectory operationsDirectory,
        FrontEndPrimaryDirectory frontEndPrimaryDirectory,
        FrontEndGuestDirectory frontEndGuestDirectory,
        FrontEnd.FrontEndDirectory frontEndDirectory,
        CallsDirectory callsDirectory,
        ToolsDirectory toolsDirectory,
        NotesAndMessagesDirectory notesAndMessagesDirectory,
        WebLinksDirectory webLinksDirectory,
        ILogger<DirectoriesMultitudeCommandingOrderRectoActionShut> logger
    )
    {
        public Dictionary<string, string> PairUpVariablesWithTheirValue(
            string fileNamePath,
            Dictionary<string, string> environmentVariablesSourceDictionary
        )
        {
            Dictionary<string, string> fileContentDictionaryToWriteToFile = [];

            const int bufferSize = 128;
            using var fileStream = File.OpenRead(fileNamePath);
            using var streamReader = new StreamReader(fileStream, Encoding.UTF8, true, bufferSize);

            while (streamReader.ReadLine() is { } line)
            {
                var brokenLine = line.Split("=");
                var key = brokenLine[0];
                var value = brokenLine[1];

                if (environmentVariablesSourceDictionary.TryGetValue(key, out var val))
                {
                    try
                    {
                        switch (key)
                        {
                            case "DIRECTORY_MANAGEMENT_EXECUTIVE_FILE_ADDRESS":
                                fileContentDictionaryToWriteToFile.Add(key, val ?? "");
                                break;
                            case "DIRECTORY_MANAGEMENT_EXECUTIVE_FILE_ADDRESS_CONTAINING_DIRECTORY":
                                {

                                    if (environmentVariablesSourceDictionary.TryGetValue("DIRECTORY_MANAGEMENT_EXECUTIVE_FILE_ADDRESS", out var directoryManagementExecutiveFileAddress))
                                    {
                                        var striped = stringHelpers.StripQuotationMarks(directoryManagementExecutiveFileAddress);
                                        var dirName = Path.GetDirectoryName(striped);
                                        if (dirName is not null)
                                        {
                                            fileContentDictionaryToWriteToFile.Add(key, dirName);
                                        }
                                    }
                                    break;
                                }
                            case "FEND_PRIMARY_ADDRESS":
                                {
                                    var frontEndPrimaryDirectoryPath = frontEndPrimaryDirectory.GetPath();
                                    var valueToWrite = stringHelpers.WrapInQuotationMarks(frontEndPrimaryDirectoryPath);
                                    fileContentDictionaryToWriteToFile.Add(key, valueToWrite ?? "");
                                    break;
                                }
                            case "FEND_GUEST_ADDRESS":
                                {
                                    var frontEndGuestDirectoryPath = frontEndGuestDirectory.GetPath();
                                    var valueToWrite = stringHelpers.WrapInQuotationMarks(frontEndGuestDirectoryPath);
                                    fileContentDictionaryToWriteToFile.Add(key, valueToWrite ?? "");
                                    break;
                                }
                            case "FEATURE_SELF_ADDRESS":
                                {
                                    var featureDirectoryPath = FeatureDirectory.GetPath();
                                    var valueToWrite = stringHelpers.WrapInQuotationMarks(featureDirectoryPath);
                                    fileContentDictionaryToWriteToFile.Add(key, valueToWrite ?? "");
                                    break;
                                }
                            case "OPERATIONS_DIRECTORY_PATH":
                                {
                                    var operationsDirectoryPath = operationsDirectory.GetPath();
                                    var valueToWrite = stringHelpers.WrapInQuotationMarks(operationsDirectoryPath);
                                    fileContentDictionaryToWriteToFile.Add(key, valueToWrite ?? "");
                                    break;
                                }
                            case "FEND_ADDRESS":
                                {
                                    var frontEndDirectoryPath = frontEndDirectory.GetPath();
                                    var valueToWrite = stringHelpers.WrapInQuotationMarks(frontEndDirectoryPath);
                                    fileContentDictionaryToWriteToFile.Add(key, valueToWrite ?? "");
                                    break;
                                }
                            case "BEND_ADDRESS":
                                {
                                    var backEndDirectoryPath = BackEndDirectory.GetPath();
                                    var valueToWrite = stringHelpers.WrapInQuotationMarks(backEndDirectoryPath);
                                    fileContentDictionaryToWriteToFile.Add(key, valueToWrite ?? "");
                                    break;
                                }
                            case "CALLS_ADDRESS":
                                {
                                    var callsDirectoryPath = callsDirectory.GetPath();
                                    var valueToWrite = stringHelpers.WrapInQuotationMarks(callsDirectoryPath);
                                    fileContentDictionaryToWriteToFile.Add(key, valueToWrite ?? "");
                                    break;
                                }
                            case "TOOLS_ADDRESS":
                                {
                                    var toolsDirectoryPath = toolsDirectory.GetPath();
                                    var valueToWrite = stringHelpers.WrapInQuotationMarks(toolsDirectoryPath);
                                    fileContentDictionaryToWriteToFile.Add(key, valueToWrite ?? "");
                                    break;
                                }
                            case "NOTES_MESSAGES_ADDRESS":
                                {
                                    var notesAndMessagesDirectoryPath = notesAndMessagesDirectory.GetPath();
                                    var valueToWrite = stringHelpers.WrapInQuotationMarks(notesAndMessagesDirectoryPath);
                                    fileContentDictionaryToWriteToFile.Add(key, valueToWrite ?? "");
                                    break;
                                }
                            case "WEB_LINKS_ADDRESS":
                                {
                                    var webLinksDirectoryPath = webLinksDirectory.GetPath();
                                    var valueToWrite = stringHelpers.WrapInQuotationMarks(webLinksDirectoryPath);
                                    fileContentDictionaryToWriteToFile.Add(key, valueToWrite ?? "");
                                    break;
                                }
                            case "STARTUP_DIRECTORY_LOCATION":
                            case "IS_OPENING_OPERATIONS_DIRECTORY":
                            case "IS_OPENING_FEND_PRIMARY_ADDRESS":
                            case "IS_OPENING_FEND_GUEST_ADDRESS":
                            case "IS_OPENING_BEND_ADDRESS":
                            case "IS_OPENING_CALLS_ADDRESS":
                            case "IS_OPENING_TOOLS_ADDRESS":
                            case "IS_OPENING_NOTES_MESSAGES_ADDRESS":
                            case "IS_OPENING_WEB_LINKS_ADDRESS":
                            case "IS_OPENING_FEATURE_SELF_ADDRESS":
                            case "IS_OPENING_FEND_ADDRESS":
                                environmentVariablesSourceDictionary.TryGetValue(key, out var val1);
                                fileContentDictionaryToWriteToFile.Add(key, val1 ?? "");
                                break;
                            default:
                                {
                                    break;
                                }
                        }
                    }
                    catch (Exception)
                    {
                        logger.LogError("DirectoriesMultitudeCommandingOrderRectoActionShut: the key could not be processed: {Key}", key);
                    }
                }
            }
            return fileContentDictionaryToWriteToFile;
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Hosting/Feature/Automations/EnvironmentVariablesTemplateFiles/DirectoriesMultitude/DirectoriesMultitudeCommandingOrderRectoActionShut.fs
namespace cafdemwimu.console.Directories.Hosting.Feature.Automations.EnvironmentVariablesTemplateFiles.DirectoriesMultitude

open System.IO
open System.Text
open System.Collections.Generic
open Microsoft.Extensions.Logging
open cafdemwimu.console.Directories.NullHelpers
open cafdemwimu.console.Directories.Hosting.Feature
open cafdemwimu.console.Directories.Hosting.Feature.BackEnd
open cafdemwimu.console.Directories.Hosting.Feature.Calls
open cafdemwimu.console.Directories.Hosting.Feature.FrontEnd
open cafdemwimu.console.Directories.Hosting.Feature.FrontEnd.FrontEndPrimary
open cafdemwimu.console.Directories.Hosting.Feature.FrontEnd.FrontEndSecondary
open cafdemwimu.console.Directories.Hosting.Feature.NotesAndMessages
open cafdemwimu.console.Directories.Hosting.Feature.Tools
open cafdemwimu.console.Directories.Hosting.Feature.WebLinks
open cafdemwimu.console.Directories.Hosting.Feature.Automations.Operations
open cafdemwimu.console.Helpers

type DirectoriesMultitudeCommandingOrderRectoActionShut
    (
        stringHelpers: StringHelpers,
        operationsDirectory: OperationsDirectory,
        frontEndPrimaryDirectory: FrontEndPrimaryDirectory,
        frontEndSecondaryDirectory: FrontEndSecondaryDirectory,
        frontEndDirectory: FrontEndDirectory,
        callsDirectory: CallsDirectory,
        toolsDirectory: ToolsDirectory,
        notesAndMessagesDirectory: NotesAndMessagesDirectory,
        webLinksDirectory: WebLinksDirectory,
        logger: ILogger<DirectoriesMultitudeCommandingOrderRectoActionShut>
    ) =
    member _.PairUpVariablesWithTheirValue(fileNamePath: string, environmentVariablesSourceDictionary: Dictionary<string, string>) =
        let fileContentDictionaryToWriteToFile = Dictionary<string, string>()

        let bufferSize = 128
        use fileStream = File.OpenRead(fileNamePath)
        use streamReader = new StreamReader(fileStream, Encoding.UTF8, true, bufferSize)

        let mutable line = streamReader.ReadLine()
        while not (isNull line) do
            let brokenLine = line.Split('=')
            let key = brokenLine.[0]
            let _value = brokenLine.[1]
            match environmentVariablesSourceDictionary.TryGetValue key with
            | true, valFromSource ->
                try
                    match key with
                    | "CAFDEM_EXECUTIVE_FILE_ADDRESS" ->
                        fileContentDictionaryToWriteToFile.Add(key, orEmpty valFromSource)
                    | "CAFDEM_EXECUTIVE_FILE_ADDRESS_CONTAINING_DIRECTORY" ->
                        match environmentVariablesSourceDictionary.TryGetValue("CAFDEM_EXECUTIVE_FILE_ADDRESS") with
                        | true, cafdemExecutiveFileAddress ->
                            let striped = stringHelpers.StripQuotationMarks(cafdemExecutiveFileAddress)
                            let dirName = Path.GetDirectoryName(striped: string)
                            if not (isNull dirName) then
                                fileContentDictionaryToWriteToFile.Add(key, dirName)
                        | _ -> ()
                    | "FEND_PRIMARY_ADDRESS" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(frontEndPrimaryDirectory.GetPath()))
                    | "FEND_SECONDARY_ADDRESS" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(frontEndSecondaryDirectory.GetPath()))
                    | "FEATURE_SELF_ADDRESS" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(FeatureDirectory.GetPath()))
                    | "OPERATIONS_DIRECTORY_PATH" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(operationsDirectory.GetPath()))
                    | "FEND_ADDRESS" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(frontEndDirectory.GetPath()))
                    | "BEND_ADDRESS" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(BackEndDirectory.GetPath()))
                    | "CALLS_ADDRESS" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(callsDirectory.GetPath()))
                    | "TOOLS_ADDRESS" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(toolsDirectory.GetPath()))
                    | "NOTES_MESSAGES_ADDRESS" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(notesAndMessagesDirectory.GetPath()))
                    | "WEB_LINKS_ADDRESS" -> fileContentDictionaryToWriteToFile.Add(key, stringHelpers.WrapInQuotationMarks(webLinksDirectory.GetPath()))
                    | "STARTUP_DIRECTORY_LOCATION"
                    | "IS_OPENING_OPERATIONS_DIRECTORY"
                    | "IS_OPENING_FEND_PRIMARY_ADDRESS"
                    | "IS_OPENING_FEND_SECONDARY_ADDRESS"
                    | "IS_OPENING_BEND_ADDRESS"
                    | "IS_OPENING_CALLS_ADDRESS"
                    | "IS_OPENING_TOOLS_ADDRESS"
                    | "IS_OPENING_NOTES_MESSAGES_ADDRESS"
                    | "IS_OPENING_WEB_LINKS_ADDRESS"
                    | "IS_OPENING_FEATURE_SELF_ADDRESS"
                    | "IS_OPENING_FEND_ADDRESS" ->
                        let _, val1 = environmentVariablesSourceDictionary.TryGetValue key
                        fileContentDictionaryToWriteToFile.Add(key, orEmpty val1)
                    | _ -> ()
                with _ ->
                    logger.LogError("DirectoriesMultitudeCommandingOrderRectoActionShut: the key could not be processed: {Key}", key)
            | _ -> ()
            line <- streamReader.ReadLine()

        fileContentDictionaryToWriteToFile
